libs/SalesforceSDK/src/com/salesforce/androidsdk/ui/OAuthWebviewHelper.java

```diff
@@ -541,8 +569,37 @@ public boolean shouldOverrideUrlLoading(WebView view, WebResourceRequest request
                         onAuthFlowComplete(tr);
                     }
                 }
+            } else {
+                // Only allow redirect for requests with authorization headers for Salesforce urls.
+                if (request.getRequestHeaders() != null && request.getRequestHeaders().containsKey("Authorization")) {
```

Comment:
Why do we have that check?
We don't want to follow redirects blindly if we are adding bearer tokens to the request (during rest calls) to avoid leaking them out but why does it matter during login?

Reply:
[iOS wasn't following any authorized redirects but changed to follow them if they are SFDC urls.](https://github.com/forcedotcom/SalesforceMobileSDK-iOS/pull/3629/files)  Just trying to reach parity.

Reply:
@bbirman this [iOS code](https://github.com/forcedotcom/SalesforceMobileSDK-iOS/pull/3629/files#diff-187e3eba81f3f8f772d3d913e610b04183eef011a4fa67e390eea88262571015) in SFNetwork only runs during REST calls not during login, or am I wrong?

Reply:
In other word, shouldn't the logic above move to RestClient.java?

Reply:
Yup on iOS it only applies for the RestClient equivalent, no changes for the web view

Reply:
😳. That makes a lot of sense.  I will fix this.